pkg/dedup/iter.go

```diff
@@ -37,9 +37,77 @@ func isCounter(f string) bool {
 	return f == "increase" || f == "rate" || f == "irate" || f == "resets"
 }
 
-func NewSeriesSet(set storage.SeriesSet, replicaLabels map[string]struct{}, f string, pushdownEnabled bool) storage.SeriesSet {
+// NewOverlapSplit splits overlapping chunks into separate series entry, so existing algorithm can work as usual.
```

Comment:
I wonder if we should be private, and the public method should be `NewSeriesSet` or `NewDedupSeriesSet`. Overlap deduplication by splitting into separate streams is mostly an implementation detail.

Reply:
Well, ideally this is temporary work - I would vote for being explicit on the caller side (: